### DISCUSSION TITLE
is there a GPL issue for Quarto-built documentation of non-GPL-compatible packages?

### DISCUSSION THREAD

**Discussion by RoyCCWang** [quarto-dev/quarto-cli#2915] [Q&A] [answered]
Thanks for developing Quarto! I understand that if one uses a GPL-licensed code editor to create proprietary software, the software does not need to adapt the GPL license because any other code editor could probably work. I just want to double-check with the Quartos community if a similar argument holds for Quartos. I have three questions.

**Q1:** If someone writes a software package that is closed source or proprietary, could they create open source  documentation for the package using Quartos without being forced to change the license on the proprietary package?

For example, the author of a closed-source web API/app library uses Quartos to build documentation. The documentation is made into a public website, and the site make available all the `*.qmd` and notebook files for the public.

**Q2:** Would the author need to indicate that the documentation, `.qmd` and notebook files, are all GPL licensed?

**Q3:** My understanding is that this should not force the closed-source web API library to adapt a GPL license, but may someone please verify for me?

My motivation for this question is to see if we as a community  can encourage proprietary package authors to use and hopefully contribute back to GPL-licensed software such as Quartos.

**Comment by jjallaire** [comment, marked as answer]
The intention is definitely that Quarto's license does not have any impact at all on the license of content created with Quarto! So please proceed with that assumption. If we need to document this more thoroughly and/or distinguish between the Quarto tools and redistributable files we will take care of that ASAP.

**Reply by RoyCCWang** [reply]
Thanks for the clarification! I think a brief paragraph under the FAQ section with your explanation is beneficial to Quartos' adaption, because I know some private sector folks unfortunately avoid working with copyleft software due to a misunderstanding of the terms.

**Reply by jjallaire** [reply]
Thanks, we've added a statement to the FAQ here: https://quarto.org/docs/faq/#can-i-use-quarto-to-develop-proprietary-content

**Reply by chriskmanx** [reply]
Hello Quarto Creators, I wanted to "briefly" chime in on this discussion. I apologize for the double posting, but I also made this comment  [here](https://github.com/quarto-dev/quarto-cli/discussions/4639). 

I spent some 10 years as an IT manager of a global hedge fund and licensing would of course, as for anyone in that role, come up where commercial software is involved. I am really quite taken by Quarto and would like to believe it will be possible to use it commercially given its GPL heritage, but I would like some clarification.  

I like the analogy by @mcanouil in respect of the Word Processor. Indeed, a strict interpretation of the line of thought by @okayucob would lead one to conclude that if I write a poem in AbiWord ( a Linux word processor distributed under the GPL ) that then the stylistic elements applied to that poem by AbiWord upon rendering the document, would prompt my poem also to be covered by the GPL --- if distributed. That's at the core of what people are concerned about here if I am not mistaken. 

Now, the GPL is intended to cover **software** and there are as of late many grey areas in the application of the GPL resulting from the evolution taking place in that space. Take the discussion around AI,  Chat-GPT, and LLaMa by MetaAI involving model weights which are not the output of a process of creative authorship covered by copyright law in the first instance. Indeed, there is a lawsuit underway in respect of Stable Diffusion, another AI package, as to whether the model weights within that software violate the copyright of the original content used in training the Stable Diffusion model. Who is right? 

Besides the editor/word processor analogy, there is perhaps another analogy that might apply here, given that Quarto is essentially a LaTeX processor or a derivative of a LaTeX processor with markdown processing added in. There are some interesting discussions on the internet, which essentially came to very different conclusions than the ones purported here. 

https://tex.stackexchange.com/questions/69007/the-gpl-and-latex-packages

The above concludes: "All cases must be licensed under the GPL, as they are all [modified versions](http://www.gnu.org/licenses/gpl-faq.html#MereAggregation), not aggregates. "

Also: 

https://opensource.stackexchange.com/questions/2735/gpl-licensed-latex-template-implications-for-resulting-work


The gentleman there has a very similar problem: combining a LaTeX template (GPL) with his proprietary work ( a thesis ) which is rendered using LaTeX. That discussion thread seems to firstly conclude that "Any derived work has to be licensed as GPL. GPL forces to make the source available. Thus, the tex source of the generated PDFs have to be made available, too" but then differentiates based on the fact that the thesis, as in my example of the poem, is data rather than code and that therefore the generated PDFs would indeed not be covered by the GPL. Now, in the case of Quarto, the inputs are code & data. Commercial authors are likely to want to protect both. This is incredibly important since if the Quarto system may be thought of as a compiler and functionalities & styles may be thought of as libraries, then any code input to that system would potentially be covered by the GPL. The word processor analogy would no longer apply. 

An argument could then be made that the code processed by Quarto is loosely coupled as it is being interpreted ( compiled )  by a forked process ( R, Python, Julia ) and that this loose coupling absolves the code from any obligations under the GPL. In reading the GPL, this only holds true if no "complex" data structures are being shared. 

This would be my first question to the creators of Quarto. Do you hold that the above argument ( loose coupling & no complex shared data structures ) holds true? 

My second question/request is that a license, GPL, or otherwise is an agreement between a content creator and a licensee. The GPL isn't a law in and of itself. Hence, the only party who could hold a licensee accountable is the content creator, not anyone else. Nobody else has recourse: Legal Framework for Open Source Software (in German) http://www.beck-shop.de/Jaeger-Metzger-Open-Source-
Software/productview.aspx?product=14503508 . Differently put, your ( Quarto creators ) stated position, therefore, is of paramount importance, as it relates to the "whole understanding" of the licensee entering a license agreement. In areas, where you indemnify your licensees, the GPL loses its strength. That would really be my second question. Are you prepared to indemnify your licensees in accordance with your stated intent, i.e. that intent being that original contenrt is not covered? 

None of the opinions stated above constitute legal advice. 

Best Regards

Chris Kohlhepp

**Comment by jjallaire** [comment]
We do indeed hold that loose coupling and no complex shared data structures holds true. Here is the section of our FAQ that attempts to make this clear: https://quarto.org/docs/faq/#can-i-use-quarto-to-develop-proprietary-content
